helper/schema/resource_diff_test.go

```diff
@@ -1237,6 +1237,36 @@ func TestGetChangedKeysPrefix(t *testing.T) {
 				"foo",
 			},
 		},
+		{
+			Name: "basic primitive diff with empty prefix",
+			Schema: map[string]*Schema{
+				"foo": {
+					Type:     TypeString,
+					Optional: true,
+					Computed: true,
+				},
+			},
+			State: &terraform.InstanceState{
+				Attributes: map[string]string{
+					"foo": "bar",
+				},
+			},
+			Config: testConfig(t, map[string]interface{}{
+				"foo": "baz",
+			}),
+			Diff: &terraform.InstanceDiff{
+				Attributes: map[string]*terraform.ResourceAttrDiff{
+					"foo": {
+						Old: "bar",
+						New: "baz",
+					},
+				},
+			},
+			Key: "",
+			ExpectedKeys: []string{
```

Comment:
Can you please also add an unchanged root attribute and changed nested attribute to ensure that they are not returned and returned respectively in this test case? Thanks!

Reply:
Done.